web/src/components/vis/PlsdaPage/PlsdaPage.vue

```diff
@@ -0,0 +1,227 @@
+<script>
+import VisTileLarge from '@/components/vis/VisTileLarge.vue';
+import LayoutGrid from '@/components/LayoutGrid.vue';
+import ScorePlot from './ScorePlot.vue';
+import LoadingsPlot from './LoadingsPlot.vue';
+import plotData from '../mixins/plotData';
+
+export default {
+  components: {
+    ScorePlot,
+    LoadingsPlot,
+    VisTileLarge,
+    LayoutGrid,
+  },
+
+  mixins: [plotData('plsda')],
+
+  props: {
+    id: {
+      type: String,
+      required: true,
+    },
+  },
+
+  data() {
+    return {
+      pcXval: '1',
+      pcYval: '2',
+      numComponentsVal: '10',
+      pcX: 1,
+      pcY: 2,
+      numComponents: 10,
+      showEllipses: true,
+      showCrosshairs: true,
+      showCutoffs: true,
+      showScore: true,
+      showLoadings: true,
+    };
+  },
+
+  computed: {
+    ready() {
+      const pcaReady = this.$store.getters.ready(this.id, 'plsda_scores');
+      const loadingsReady = this.$store.getters.ready(this.id, 'plsda_loadings');
+      return pcaReady && loadingsReady;
+    },
+    loadings() {
+      if (this.plot?.data?.loadings) {
+        return this.plot.data.loadings;
+      }
+      return [];
+    },
+    pcCoords() {
+      if (this.plot?.data?.scores?.x) {
+        return this.plot.data.scores.x;
+      }
+      return [];
+    },
+    eigenvalues() {
+      if (this.plot?.data?.scores?.sdev) {
+        return this.plot.data.scores.sdev;
+      }
+      return [];
+    },
+    rowLabels() {
+      if (this.dataset?.row?.data) {
+        return this.dataset.row.data.filter((row) => row.row_type === 'sample').map((row) => row.row_name);
+      }
+      return [];
+    },
+    groupLabels() {
+      if (this.dataset?.validatedGroups?.data) {
+        const groupColumnName = this.dataset.validatedGroups.columnNames[0];
+        const groupLabels = {};
+        groupLabels[groupColumnName] = this.dataset.validatedGroups.data.map((group) => group);
+        return groupLabels;
+      }
+      return {};
+    },
+    columns() {
+      if (this.dataset?.column?.data) {
+        return this.dataset.column.data;
+      }
+      return [];
+    },
+
+    groupLevels() {
+      if (this.dataset?.groupLevels) {
+        return this.dataset.groupLevels;
```

Comment:
is `groupLevels` just an array of the sample groups in a dataset?

Reply:
Yes, I believe so.